### PR TITLE
Voluntary exit CLI/Client integration

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

The aim of this PR is to provide full integration between the CLI command for performing an exit, the validator and the gRPC validator client.

**Which issues(s) does this PR fix?**

Part of #6882 

**Other notes for review**

N/A

### CODE REVIEW

File: validator/accounts/v2/accounts_exit.go [hunk 1]
```diff
@@ -101,7 +106,42 @@ func ExitAccountsCli(cliCtx *cli.Context, r io.Reader) error {
...
+	if err != nil {
+		return errors.Wrapf(err, "could not dial endpoint %s", flags.BeaconRPCProviderFlag.Name)
+	}
+	validatorClient := ethpb.NewBeaconNodeValidatorClient(conn)
+
+	var rawNotExitedKeys [][]byte
+	for _, key := range rawPublicKeys {
+		if err := client.ProposeExit(cliCtx.Context, validatorClient, keymanager, key); err != nil {
+			rawNotExitedKeys = append(rawNotExitedKeys, key)
+			log.WithError(err).Errorf("voluntary exit failed for account %s", key)
```
Comment (rkapka) [inline validator/accounts/v2/accounts_exit.go:129]: Display the formatted key.

Comment (terencechain) [reply validator/accounts/v2/accounts_exit.go:129]: please display key as hex string, check proposer and attester .go for consistency

File: validator/accounts/v2/accounts_exit.go [hunk 2]
```diff
@@ -101,7 +106,42 @@ func ExitAccountsCli(cliCtx *cli.Context, r io.Reader) error {
...
+	for _, notExited := range rawNotExitedKeys {
+		for i, key := range rawPublicKeys {
+			if bytes.Equal(notExited, key) {
+				formattedExitedKeys = append(formattedExitedKeys, formattedPubKeys[i])
+				break
+			}
+		}
+	}
+
+	log.WithField("publicKeys", strings.Join(formattedExitedKeys, ", ")).
```
Comment (terencechain) [inline validator/accounts/v2/accounts_exit.go:142]: same, please check consistency for logging key

File: validator/accounts/v2/accounts_exit_test.go [hunk 3]
```diff
@@ -81,3 +81,5 @@ func TestExitAccountsCli_EmptyWalletReturnsError(t *testing.T) {
 	err = ExitAccountsCli(cliCtx, os.Stdin)
 	assert.ErrorContains(t, "wallet is empty, no accounts to perform voluntary exit", err)
 }
+
+// TODO: Write an integration test?
```
Comment (terencechain) [inline validator/accounts/v2/accounts_exit_test.go:85]: yes, please add test

**Comment by rkapka** [comment]
`TestExitAccountsCli_Ok()` fails because of these two lines:
https://github.com/prysmaticlabs/prysm/blob/c3eef978decc74c5100dc8b808b7afab8fd55801/validator/accounts/v2/accounts_exit.go#L168-L169
I am `New`ing the clients and because of this I cannot use `gomock` to mock them. Is there any standard pattern that should be used here? Initialize them inside `main()` and pass into `ExitAccountsCli()` so that I can pass a mock implementation in tests?

**Review by rauljordan — APPROVED** [review]
Nice work on this one!

**Pull request merged**